pilot/pkg/networking/plugin/plugin.go

```diff
@@ -34,15 +34,15 @@ const (
 	ListenerTypeHTTP
 )
 
-// CallbackListenerInputParams is a set of values passed to On*Listener callbacks. Not all fields are guaranteed to
+// InputParams is a set of values passed to Plugin callback methods. Not all fields are guaranteed to
 // be set, it's up to the callee to validate required fields are set and emit error if they are not.
 // These are for reading only and should not be modified.
-type CallbackListenerInputParams struct {
-	// ListenerType is the type of listener (TCP, HTTP etc.)
+type InputParams struct {
```

Comment:
that sounds fine, the only thing is that it's now an input type to both OnListener and OnRoute (it's arm waving that you shouldn't set any field that doesn't apply to whatever callback to avoid defining nearly the same struct multiple times).
 i can throw out a few alternatives:
1. create different types for OnListener / OnRoute. actually i'm also a bit troubled by the vagueness of what can be expected to be set in the callbacks. there are a lot of implicit rules that are not clear from the API. making each type separate for each flavor of callback would make this much clearer at the expense of having lots of very similar types.
2. go back to having long list of params per callback. this is going to get messy, but less duplication 
3. some kind of blend of 1&2 where base params are distilled into something that's really just plugin.BaseParams and is guaranteed to be always set and the rest are per plugin. 
4. keep things as is

i guess i'd vote for 3 if there's really a "base params" - i'm just not sure if there's anything other than env that's common to all in which case it collapses to option 2 - long lists of function arguments. it's going to be up to seven arguments which is pushing it.

Reply:
I agree 3 is probably the best in this case. Either way, not blocking on this PR IMO.